rust/src/pgsql/pgsql.rs

```diff
@@ -341,16 +368,23 @@ impl PgsqlState {
             );
             match PgsqlState::state_based_req_parsing(self.state_progress, start) {
                 Ok((rem, request)) => {
-                    sc_app_layer_parser_trigger_raw_stream_reassembly(flow, Direction::ToServer as i32);
                     start = rem;
+                    let mut temp_state = PgsqlStateProgress::IdleState;
                     if let Some(state) = PgsqlState::request_next_state(&request) {
                         self.state_progress = state;
+                        temp_state = state;
                     };
-                    let tx_completed = self.is_tx_completed();
+                    let tx_completed = self.is_tx_completed(Direction::ToServer);
                     if let Some(tx) = self.find_or_create_tx() {
                         tx.request = Some(request);
                         if tx_completed {
-                            tx.tx_state = PgsqlTransactionState::ResponseDone;
+                            if temp_state == PgsqlStateProgress::ConnectionTerminated || temp_state == PgsqlStateProgress::CancelRequestReceived { 
```

Comment:
Would you have some state machine schema/drawing ? like https://datatracker.ietf.org/doc/html/rfc9113#name-stream-states

Reply:
Back when I was studying the PostgreSQL, I found two:
<img width="1341" alt="statemachine" src="https://github.com/user-attachments/assets/6989c486-6b2b-4e96-ae54-230f65638240">
![PGConnection StateMachineOutdated](https://github.com/user-attachments/assets/af4f7864-5f56-4f27-8311-36e5429523f3)

Do you suggest that I create one?

Reply:
Could you just reference these ?
Where do they come from ?

Reply:
I'll add a reference to the first image. Oddly, I can't find where the other comes from - not even with Google search. And I don't think I created it. '-'